UndertaleModLib/Models/UndertaleShader.cs

```diff
@@ -528,10 +528,20 @@ public void WriteData(UndertaleWriter writer)
             writer.Write(Data);
         }
 
-        public void ReadData(UndertaleReader reader, int length)
+        public void ReadData(UndertaleReader reader, int length, bool isLast = false)
         {
-            if (_Length != 0 && _Length != length)
-                throw new UndertaleSerializationException("Failed to compute length of shader data");
+            if (_Length != 0) // _Length is the "expected" length here
+            {
+                if (_Length > length)
+                    throw new UndertaleSerializationException("Failed to compute length of shader data: instructed to read less data than expected.");
+                else if (_Length < length)
+                {
+                    if (isLast) // Normal for the last element due to chunk padding, just trust the system
```

Comment:
Is there no way to detect how much padding we should have? would be nicer.

Reply:
I haven't done extensive testing with this. I think the maximum padding would be 15 bytes but I don't know everything about padding and kind of figured we try to work with whatever padding the file comes with.

Reply:
Well, if you know how things are aligned on, you can determine that by checking the current reader position.
i.e. if things are always 16-aligned, and your current position is on 30, you'd have 2 padding. If your position is on 20, you'd have 12 padding etc.